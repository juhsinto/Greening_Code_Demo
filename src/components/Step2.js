import React from "react";

import {useForm} from "react-hook-form";
import { useDispatch, useSelector } from "react-redux";
import { useHistory } from "react-router-dom";
import { updatePage2, updateFormProgress } from "../rootSlice";
import {Result} from "./Result"
import {Header} from "./Header"

import {Jumbotron,Container,InputGroup, Form, FormLabel, Button} from 'react-bootstrap';


export const Step2 = () => {

    const dispatch = useDispatch()
    const history = useHistory()
    

    const vc_funded = useSelector(state => state.page2.vc_funded)
    const investment = useSelector(state => state.page2.investment)
    
    const physical = useSelector(state => state.page2.physical)
    const construction = useSelector(state => state.page2.construction)

    const project_progress = useSelector(state => state.page2.project_progress)
    
    const {register, handleSubmit} = useForm({defaultValues: {vc_funded, investment, physical, construction, project_progress}});

    const onSubmit = (data) => {
        
        dispatch(
            updatePage2({
              ...data,
            })
          );
        dispatch(updateFormProgress(50));
        history.push("./step3")
    }

    const goBack = () => {
        dispatch(updateFormProgress(0));
        history.push("./");
    };

    return (
        <Container className="p-3">
            <Jumbotron>
                <Header />
                <form onSubmit={handleSubmit(onSubmit)}>
                    <div>
                        <label htmlFor="page1">Page 2 Details:</label>
                        <br />
                        
                        

                        <InputGroup className="p-3">
                            <Form.Group>
                            <FormLabel>Funding Sources</FormLabel>
                                <Form.Check type="checkbox" label="VC Funded" 
                                name="vc_funded" id="vc_funded" 
                                ref={register}
                                />
                                <Form.Check type="checkbox" label="Investment Available" 
                                name="investment" id="investment" 
                                ref={register}
                                />
                            </Form.Group>
                        </InputGroup> 

                        <InputGroup className="p-3">
                            <Form.Group>
                            <FormLabel>Project Type</FormLabel>
                                <Form.Check type="checkbox" label="Physical Project" 
                                name="physical" id="physical" 
                                ref={register}
                                />
                                <Form.Check type="checkbox" label="Construction Project" 
                                name="construction" id="construction" 
                                ref={register}
                                />
                            </Form.Group>
                        </InputGroup> 

                        <FormLabel>Project Progress %</FormLabel>
                        <Form.Group>
                            <Form.Control name="project_progress" id="project_progress"  type="range" ref={register} />
                        </Form.Group>

                        {/* was trying to do radio type but wasn't able to get the data satisfactorily */}
                        {/* <InputGroup className="p-3">
                            <fieldset>
                            <Form.Group as={Row} 
                            <Form.Label as="legend" column sm={2} >
                                Radios
                            </Form.Label>
                            <Col sm={10}>
                                <Form.Check
                                type="radio"
                                label="first radio"
                                name="one"
                                id="formHorizontalRadios1"
                                ref={register}>
                                />
                                <Form.Check
                                type="radio"
                                label="second radio"
                                name="two"
                                id="formHorizontalRadios2"
                                ref={register}>
                                />
                                <Form.Check
                                type="radio"
                                label="third radio"
                                name="three"
                                id="formHorizontalRadios3"
                                ref={register}>
                                />
                            </Col>
                            </Form.Group>
                            </fieldset>
                        </InputGroup>  */}
                        

                        <Button onClick={goBack}>Previous</Button>{' '}
                        <Button type="submit">Next</Button>
                        </div>
                </form>
            </Jumbotron>
            <Result />
        </Container>
    )
}

export default Step2;
