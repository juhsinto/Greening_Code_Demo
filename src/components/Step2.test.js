import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Step2 from './Step2';
import { store } from '../store';

it('renders Step2 correctly', () => {
  const tree = renderer.create(<Provider store={store}><Step2 /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});