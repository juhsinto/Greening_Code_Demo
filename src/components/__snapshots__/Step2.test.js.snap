// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Step2 correctly 1`] = `
<div
  className="p-3 container"
>
  <div
    className="jumbotron"
  >
    <h1
      className="header"
    >
      A simple Questionnaire
    </h1>
    <form
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="page1"
        >
          Page 2 Details:
        </label>
        <br />
        <div
          className="p-3 input-group"
        >
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Funding Sources
            </label>
            <div
              className="form-check"
            >
              <input
                className="form-check-input"
                disabled={false}
                id="vc_funded"
                name="vc_funded"
                type="checkbox"
              />
              <label
                className="form-check-label"
                htmlFor="vc_funded"
                title=""
              >
                VC Funded
              </label>
            </div>
            <div
              className="form-check"
            >
              <input
                className="form-check-input"
                disabled={false}
                id="investment"
                name="investment"
                type="checkbox"
              />
              <label
                className="form-check-label"
                htmlFor="investment"
                title=""
              >
                Investment Available
              </label>
            </div>
          </div>
        </div>
        <div
          className="p-3 input-group"
        >
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Project Type
            </label>
            <div
              className="form-check"
            >
              <input
                className="form-check-input"
                disabled={false}
                id="physical"
                name="physical"
                type="checkbox"
              />
              <label
                className="form-check-label"
                htmlFor="physical"
                title=""
              >
                Physical Project
              </label>
            </div>
            <div
              className="form-check"
            >
              <input
                className="form-check-input"
                disabled={false}
                id="construction"
                name="construction"
                type="checkbox"
              />
              <label
                className="form-check-label"
                htmlFor="construction"
                title=""
              >
                Construction Project
              </label>
            </div>
          </div>
        </div>
        <label
          className="form-label"
        >
          Project Progress %
        </label>
        <div
          className="form-group"
        >
          <input
            className="form-control-range"
            id="project_progress"
            name="project_progress"
            type="range"
          />
        </div>
        <button
          className="btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Previous
        </button>
         
        <button
          className="btn btn-primary"
          disabled={false}
          type="submit"
        >
          Next
        </button>
      </div>
    </form>
  </div>
  <p>
    JSON data:
  </p>
  <pre>
    {
  "page1": {},
  "page2": {},
  "page3": {},
  "formProgress": 0
}
  </pre>
  <p>
    Form Progress:
  </p>
  <div
    className="progress"
  >
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={0}
      className="progress-bar"
      role="progressbar"
      style={
        Object {
          "width": "0%",
        }
      }
    >
      0%
    </div>
  </div>
</div>
`;
